### DISCUSSION TITLE
Make langfuse deployable to AWS Lambda

### DISCUSSION THREAD

**Discussion by MeddahAbdellah** [langfuse/langfuse#2877] [Ideas]
### Describe the feature or potential improvement

Langfuse today must run on a dedicated server or in a kube cluster to be able to scale it down to 0 when it's not used. It would be cool to be able to deploy it to an aws lambda so it's only available when needed, also it eases the deployment.

### Additional information

I'd be happy to add the feature myself

**Comment by marcklingen** [comment]
Unfortunately, this will not be possible as Langfuse will require more and more asynchronous operations to fully work.

**Reply by MeddahAbdellah** [reply]
Thank you @marcklingen for your response.I am refering to the nextJs app, It's a simple SSR app I don't see why it can't be on a lambda, even if it does asynchronous operations.

**Reply by marcklingen** [reply]
More and more features will depend on the `worker` container which is introduced in `v3` which will make Langfuse more scalable, reliable, and enable features such as llm-as-a-judge/automated-evaluation, alerts, webhooks, and other async workflows

**Reply by MeddahAbdellah** [reply]
I am mainly referring to the `web` container that has the app that shows the data, isn't that app independent from the `worker`?

**Reply by marcklingen** [reply]
Currently yes, will change in upcoming v3 change as the api will only log new data to a queue which is then persisted by the worked to increase throughput of the overall application

**Comment by MeddahAbdellah** [comment]
Thank you for your answers, I understand why it's not possible. ( sorry for the late response, was on holiday ).

I am closing the Discussion as it has been concluded that due to langfuse roadmap, turning the app into a serverless app is undesired.

**Reply by marcklingen** [reply]
no worries

You should be able to scale the langfuse web container to zero though if you do not care about cold-starts, e.g. on cloud run
